trin-core/src/portalnet/overlay.rs

```diff
@@ -258,56 +258,72 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
+
+                        self.init_find_content_stream(enr, conn_id).await
+                    }
+                }
+            }
+            Ok(_) => Err(OverlayRequestError::InvalidResponse),
+            Err(error) => Err(error),
+        }
+    }
+
+    /// Initialize FIndContent uTP stream with remote node
+    async fn init_find_content_stream(
+        &self,
+        enr: Enr,
+        conn_id: u16,
+    ) -> Result<Content, OverlayRequestError> {
+        let utp_request = UtpListenerRequest::FindContentStream(conn_id);
+        if let Err(err) = self.utp_listener_tx.send(utp_request) {
+            return Err(OverlayRequestError::UtpError(format!(
+                "Unable to send uTP FindContent stream request: {err}"
+            )));
+        }
+
+        // initiate the connection to the acceptor
+        let (tx, rx) = tokio::sync::oneshot::channel::<anyhow::Result<UtpSocket>>();
+
+        let _ = self
+            .utp_listener_tx
+            .send(UtpListenerRequest::Connect(conn_id, enr.node_id(), tx));
+
+        match rx.await {
+            Ok(conn) => {
+                match conn {
+                    Ok(mut conn) => {
+                        let mut result = Vec::new();
+                        // Loop and receive all DATA packets, similar to `read_to_end`
+                        loop {
+                            let mut buf = [0; BUF_SIZE];
+                            match conn.recv_from(&mut buf).await {
+                                Ok((0, _)) => {
+                                    break;
+                                }
+                                Ok((bytes, _)) => {
+                                    result.extend_from_slice(&mut buf[..bytes]);
+                                }
+                                Err(err) => {
+                                    warn!("Unable to receive content via uTP: {err}");
+                                    return Err(OverlayRequestError::UtpError(err.to_string()));
                                 }
-                                Ok(Content::Content(VariableList::from(result)))
-                            }
-                            Err(err) => {
-                                warn!("Unable to initiate uTP stream with remote node {err}");
-                                Err(OverlayRequestError::UtpError(err.to_string()))
                             }
                         }
+                        Ok(Content::Content(VariableList::from(result)))
+                    }
+                    Err(err) => {
+                        warn!("Unable to initiate uTP stream with remote node {err}");
+                        Err(OverlayRequestError::UtpError(err.to_string()))
                     }
                 }
             }
-            Ok(_) => Err(OverlayRequestError::InvalidResponse),
-            Err(error) => Err(error),
+            Err(err) => {
+                warn!("Unable to initiate uTP stream with remote node {err}");
```

Comment:
If we start seeing this log, it would be nice to immediately be able to tell whether it's this line or 318, so maybe disambiguate the two somehow? Bonus points for further explaining the kind of failure it might be in the log.